drake/systems/trajectories/isCollisionFree_mex.cpp

```diff
@@ -0,0 +1,200 @@
+
+Inputs:
+x: state from which funnel is executed (only the cyclic dimensions matter, really)
+
+forest: cell array containing vertices of obstacles
+
+funnelLibrary: struct containing funnel library
+	funnelLibrary(*).xyz: xyz positions at time points (double 3 X N)
+	funnelLibrary(*).cS: cholesky factorization of projection of S matrix (cell array)
+
+funnelIdx: index of funnel to be checked
+
+Output: Whether funnel is collision free (boolean)
+
+Author: Anirudha Majumdar
+Date: Nov 12 2013
+*/
+
+// Mex stuff
+#include <mex.h>
+#include <math.h>
+#include <matrix.h>
+
+
+// Internal access to bullet
+#include "LinearMath/btTransform.h"
+
+#include "BulletCollision/CollisionShapes/btSphereShape.h"
+#include "BulletCollision/CollisionShapes/btBoxShape.h"
+#include "BulletCollision/CollisionShapes/btConvexHullShape.h"
+// #include <iostream>
+
+#include "BulletCollision/NarrowPhaseCollision/btGjkPairDetector.h"
+#include "BulletCollision/NarrowPhaseCollision/btPointCollector.h"
+#include "BulletCollision/NarrowPhaseCollision/btVoronoiSimplexSolver.h"
+#include "BulletCollision/NarrowPhaseCollision/btConvexPenetrationDepthSolver.h"
+#include "BulletCollision/NarrowPhaseCollision/btGjkEpaPenetrationDepthSolver.h"
+#include "LinearMath/btTransformUtil.h"
+
+using namespace std;
+
+// Set solvers for bullet
+static btVoronoiSimplexSolver sGjkSimplexSolver;
+static btGjkEpaPenetrationDepthSolver epaSolver;
+
+/*Sphere of radius r representing the point. 
+We could probably make the radius 0 and be ok, but I'm not sure if bullet expects things to be non-degenrate.
+*/
+static const double g_radius = 1.0; // Don't change: leave at 1.0
+static btSphereShape* g_point = new btSphereShape(g_radius); 
+
+double ptToPolyBullet(double *vertsPr, size_t nRows, size_t nCols){
+
+	// Initialize polytope object with single point
+	btConvexHullShape polytope(btVector3(vertsPr[0],vertsPr[1],vertsPr[2]), 1);
+
+	// Add rest of the points (note the indexing starts from 1 on the loop)
+	for(int i=1;i<nCols;i++){
+			polytope.addPoint(btVector3(vertsPr[i*nRows],vertsPr[i*nRows+1],vertsPr[i*nRows+2]));
+
+	}
+
+	 // Assign elements of verts (input) to polytope
+	btTransform tr;
+	btGjkPairDetector::ClosestPointInput input; 
+	tr.setIdentity();
+	input.m_transformA = tr;
+	input.m_transformB = tr;
+
+	btGjkPairDetector convexConvex(g_point,&polytope,&sGjkSimplexSolver,&epaSolver);
+		
+	// Output
+	btPointCollector gjkOutput; 
+
+	convexConvex.getClosestPoints(input, gjkOutput, 0);
+	
+    return gjkOutput.m_distance + g_radius + CONVEX_DISTANCE_MARGIN;
+
+}
+
+
+/* Shift and transform vertices
+ */
+double *shiftAndTransform(double *verts, double *vertsT, const mxArray *x, mxArray *x0, int k, mxArray *cSk, size_t nRows, size_t nCols)
+{
+    /* This can and maybe should be sped up. For example, we know that cSk is upper triangular.*/
+    double *dcSk = mxGetPr(cSk);
+    double *dx0 = mxGetPr(x0);
+    double *dx = mxGetPr(x);
```

Comment:
These should really be mxGetPrSafe
Nonenof the mex code in this request really uses our safety wrappers.